packages/@lwc/engine-core/src/framework/def.ts

```diff
@@ -156,9 +155,9 @@ function createComponentDef(Ctor: LightningElementConstructor): ComponentDef {
     errorCallback = errorCallback || superDef.errorCallback;
     render = render || superDef.render;
 
-    let preferNativeShadow = superDef.preferNativeShadow;
-    if (!isUndefined(ctorPreferNativeShadow)) {
-        preferNativeShadow = Boolean(ctorPreferNativeShadow);
+    let shadowSupportMode = superDef.shadowSupportMode;
+    if (!isUndefined(ctorShadowMode)) {
+        shadowSupportMode = ctorShadowMode;
```

Comment:
@ekashida Please see my comment here (https://github.com/salesforce/lwc-rfcs/pull/49#discussion_r690802338). I'm concerned about subclasses not being able to fully override their superclass's `shadowMode`.

Also we should probably add some tests that confirm that subclasses inherit `renderMode`/`shadowMode` from their parent. I confirmed manually that this is the case, but I'm not sure we have tests for it.

Reply:
I added some tests for `renderMode`; this is the kind of thing I would like to test for `shadowSupportMode`: https://github.com/salesforce/lwc/pull/2462

Reply:
Will add those separately